pandas/tests/indexes/datetimes/test_date_range.py

```diff
@@ -544,89 +544,154 @@ def test_range_tz_dateutil(self):
-        left = date_range(begin, end, closed="left", freq=freq)
-        right = date_range(begin, end, closed="right", freq=freq)
-        expected_left = left
-        expected_right = right
-
-        if end == closed[-1]:
-            expected_left = closed[:-1]
-        if begin == closed[0]:
-            expected_right = closed[1:]
-
-        tm.assert_index_equal(expected_left, left)
-        tm.assert_index_equal(expected_right, right)
+        result_range = date_range(
+            begin, end, inclusive=inclusive_endpoints_fixture, freq=freq
+        )
+        both_range = date_range(begin, end, inclusive="both", freq=freq)
+
+        left_match = begin == both_range[0]
+        right_match = end == both_range[-1]
+
+        # Scenarios where datetimes are to be removed
+        if inclusive_endpoints_fixture == "left" and right_match:
+            expected_range = both_range[:-1]
+        elif inclusive_endpoints_fixture == "right" and left_match:
+            expected_range = both_range[1:]
+        elif inclusive_endpoints_fixture == "neither" and left_match and right_match:
+            expected_range = both_range[1:-1]
+        elif inclusive_endpoints_fixture == "neither" and right_match:
+            expected_range = both_range[:-1]
+        elif inclusive_endpoints_fixture == "neither" and left_match:
+            expected_range = both_range[1:]
+        elif inclusive_endpoints_fixture == "both":
+            expected_range = both_range[:]
+        else:
+            expected_range = both_range[:]
+
+        tm.assert_index_equal(expected_range, result_range)
 
-    def test_range_closed_with_tz_aware_start_end(self):
+    @pytest.mark.parametrize("freq", ["1D", "3D", "2M", "7W", "3H", "A"])
+    def test_range_closed_with_tz_aware_start_end(
+        self, freq, inclusive_endpoints_fixture
+    ):
         # GH12409, GH12684
         begin = Timestamp("2011/1/1", tz="US/Eastern")
         end = Timestamp("2014/1/1", tz="US/Eastern")
 
-        for freq in ["1D", "3D", "2M", "7W", "3H", "A"]:
-            closed = date_range(begin, end, closed=None, freq=freq)
-            left = date_range(begin, end, closed="left", freq=freq)
-            right = date_range(begin, end, closed="right", freq=freq)
-            expected_left = left
-            expected_right = right
-
-            if end == closed[-1]:
-                expected_left = closed[:-1]
-            if begin == closed[0]:
-                expected_right = closed[1:]
-
-            tm.assert_index_equal(expected_left, left)
-            tm.assert_index_equal(expected_right, right)
+        result_range = date_range(
+            begin, end, inclusive=inclusive_endpoints_fixture, freq=freq
+        )
+        both_range = date_range(begin, end, inclusive="both", freq=freq)
+
+        left_match = begin == both_range[0]
+        right_match = end == both_range[-1]
+
+        if inclusive_endpoints_fixture == "left" and right_match:
```

Comment:
consider making this a function and share among these tests

Reply:
I defined the function on the top of the test_date_range.py file, does it belong elsewhere?